### PR TITLE
Bugfix: Restart issue

### PR DESCRIPTION
**Pull request by helto4real**
## Proposed change

Fixes the restart problem where apps is not loaded the second time due to cahce was not cleared

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: src/Runtime/NetDaemon.Runtime/Internal/AppStateManager.cs [hunk 1]
```diff
@@ -59,6 +59,12 @@ public async Task SaveStateAsync(string applicationId, ApplicationState state)
         }
     }
 
+    internal Task InitializeAsync()
```
Comment (FrankBakkerNl) [inline src/Runtime/NetDaemon.Runtime/Internal/AppStateManager.cs:62]: It seems weird to have InitializeAsync and Initialize wile InitializeAsync does not even do any async stuff

Comment (helto4real) [reply src/Runtime/NetDaemon.Runtime/Internal/AppStateManager.cs:62]: Yea I forgot to change this. I had it on the interface at first but realize I did not need it when I could just put it on the class itself. I will change it.

Comment (helto4real) [reply src/Runtime/NetDaemon.Runtime/Internal/AppStateManager.cs]: Saw now. Must been drunk or smth. Haha good catch.

File: src/Runtime/NetDaemon.Runtime/Internal/AppStateManager.cs [hunk 2]
```diff
@@ -122,6 +123,11 @@ public static string ToSafeHomeAssistantEntityIdFromApplicationId(string applica
         return $"input_boolean.netdaemon_{stringBuilder.ToString().ToLowerInvariant()}";
     }
 
+    private void ClearExistingCacheOnNewConnection()
```
Comment (FrankBakkerNl) [inline src/Runtime/NetDaemon.Runtime/Internal/AppStateManager.cs:126]: suggsetion: This could be inlined the separate methid here does not add much

**Review by FrankBakkerNl — APPROVED** [review]

**Pull request merged**